### PR TITLE
CLOUDP-149877: Add connection string creation for serverless instance

### LINKED ISSUE #791: Secret with connection secrets is not created for serverless instance
**What did you do to encounter the bug?**

Apply the following helm template

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: testing
  labels:
    atlas.mongodb.com/type: "credentials"
type: Opaque
stringData:
  password: "OENkWWcxR3JleVZucGNsU1Z5M3lxSkpjTllBejhVeWM="
---
apiVersion: atlas.mongodb.com/v1
kind: AtlasDatabaseUser
metadata:
  name: atlas-deployment-user-testing
  labels:
    helm.sh/chart: testing-0.0.1_replaced.by.yq.in.pipeline
    app.kubernetes.io/name: testing
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "1.0.0"
    app.kubernetes.io/managed-by: Helm
spec:
  username: tester
  passwordSecretRef:
    name: tester-secret
  projectRef:
    name: mongodb-atlas-project
    namespace: mongodb-atlas
  roles:
    - databaseName: testdb
      roleName: readWrite
---
apiVersion: atlas.mongodb.com/v1
kind: AtlasDeployment
metadata:
  name: atlas-deployment
  labels:
    helm.sh/chart: testing-0.0.1_replaced.by.yq.in.pipeline
    app.kubernetes.io/name: testing
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "1.0.0"
    app.kubernetes.io/managed-by: Helm
spec:
  projectRef:
    name: mongodb-atlas-project
    namespace: mongodb-atlas
  backupRef:
    name: ''
    namespace: ''
  serverlessSpec:
    name: testdeployment
    providerSettings:
      providerName: AZURE
      regionName: EUROPE_WEST
```

**What did you expect?**
- [OK] AtlasDeployment created in k8s
  - [OK] Deployment created in Atlas Cloud
- [OK] AtlasDatabaseUser created in k8s
  - [OK] User created in Atlas Cloud
- **[NOK]** Secret containing the [connection secrets](https://www.mongodb.com/docs/atlas/reference/atlas-operator/atlasdatabaseuser-custom-resource/#connection-secrets) o be created in k8s

**What happened instead?**
Everything else worked as expected, just the secret containing the connection secrets was not created.

**NOTE**
After trying the same template and changing the `serverlessSpec` to a `deploymentSpec` we see the `secret` getting created.

**Screenshots**

**Operator Information**
 - Operator Version: 1.5.0

**Kubernetes Cluster Information**
 - Distribution
 - Version: 1.24.3
 - Image Registry location (quay, or an internal registry)

**Additional context**

If possible, please include:
 - `kubectl describe` output
 - yaml definitions for your objects
 - log files for the operator and database pods

### PR DESCRIPTION
**Pull request by Sugar-pack**
### All Submissions:

closes #791 

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/controller/connectionsecret/connectionsecrets.go [hunk 1]
```diff
@@ -56,6 +56,28 @@ func CreateOrUpdateConnectionSecrets(ctx *workflow.Context, k8sClient client.Cli
...
+	for _, c := range serverlessDeployments.Results {
+		found := false
+		for _, regularDeployment := range deployments {
+			if regularDeployment.Name == c.Name {
+				found = true
+				break
+			}
+		}
+
+		if !found {
```
Comment (fabritsius) [inline pkg/controller/connectionsecret/connectionsecrets.go:73]: if `not found` ?

Comment (Sugar-pack) [reply pkg/controller/connectionsecret/connectionsecrets.go:74]: Sure. New code handle serverless deployments that were not fetched thought v1(regular) and v1.5 API (advanced).

Comment (fabritsius) [reply pkg/controller/connectionsecret/connectionsecrets.go:74]: I see :)

**Review by fabritsius — APPROVED** [review]
Looks Great 👍

**Review by helderjs — APPROVED** [review]
Looks good!

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**